docs/scenarios/library-authoring/trimming.md

```diff
@@ -0,0 +1,177 @@
+        "System.Threading.Timer": "4.0.1-rc2-24027",
+        "System.Xml.ReaderWriter": "4.0.11-rc2-24027",
+        "System.Xml.XDocument": "4.0.11-rc2-24027"
+    }
+}
+```
+
+Next, you would copy over those dependencies into the `dependencies` section of the library's `project.json` file:
+
+```json
+{
+    "dependencies":{
+        "Microsoft.NETCore.Platforms": "1.0.1-rc2-23931",
+        "Microsoft.NETCore.Runtime": "1.0.2-rc2-23931",
+        "Microsoft.Win32.Primitives": "4.0.1-rc2-23931",
+        "System.AppContext": "4.1.0-rc2-23931",
+        "System.Collections": "4.0.11-rc2-23931",
+        "System.Collections.Concurrent": "4.0.12-rc2-23931",
+        "System.Console": "4.0.0-rc2-23931",
+        "System.Diagnostics.Debug": "4.0.11-rc2-23931",
+        "System.Diagnostics.Tools": "4.0.1-rc2-23931",
+        "System.Diagnostics.Tracing": "4.1.0-rc2-23931",
+        "System.Globalization": "4.0.11-rc2-23931",
+        "System.Globalization.Calendars": "4.0.1-rc2-23931",
+        "System.IO": "4.1.0-rc2-23931",
+        "System.IO.Compression": "4.1.0-rc2-23931",
+        "System.IO.Compression.ZipFile": "4.0.1-rc2-23931",
+        "System.IO.FileSystem": "4.0.1-rc2-23931",
+        "System.IO.FileSystem.Primitives": "4.0.1-rc2-23931",
+        "System.Linq": "4.1.0-rc2-23931",
+        "System.Net.Http": "4.0.1-rc2-23931",
+        "System.Net.Primitives": "4.0.11-rc2-23931",
+        "System.Net.Sockets": "4.1.0-rc2-23931",
+        "System.ObjectModel": "4.0.12-rc2-23931",
+        "System.Reflection": "4.1.0-rc2-23931",
+        "System.Reflection.Extensions": "4.0.1-rc2-23931",
+        "System.Reflection.Primitives": "4.0.1-rc2-23931",
+        "System.Resources.ResourceManager": "4.0.1-rc2-23931",
+        "System.Runtime": "4.1.0-rc2-23931",
+        "System.Runtime.Extensions": "4.1.0-rc2-23931",
+        "System.Runtime.Handles": "4.0.1-rc2-23931",
+        "System.Runtime.InteropServices": "4.1.0-rc2-23931",
+        "System.Runtime.InteropServices.RuntimeInformation": "4.0.0-rc2-23931",
+        "System.Runtime.Numerics": "4.0.1-rc2-23931",
+        "System.Text.Encoding": "4.0.11-rc2-23931",
+        "System.Text.Encoding.Extensions": "4.0.11-rc2-23931",
+        "System.Text.RegularExpressions": "4.0.12-rc2-23931",
+        "System.Threading": "4.0.11-rc2-23931",
+        "System.Threading.Tasks": "4.0.11-rc2-23931",
+        "System.Threading.Timer": "4.0.1-rc2-23931",
+        "System.Xml.ReaderWriter": "4.0.11-rc2-23931",
+        "System.Xml.XDocument": "4.0.11-rc2-23931"
+    },
+    "frameworks": {
+        "netstandard1.0": {}
+    }
+}
+```
+
+Woah!  That's quite a lot of packages, many of which certainly aren't necessary for extending collection types.  In this hypothetical situation, you may use a tool such as [ILSpy](ilspy) or [.NET Reflector](reflector) to identify which packages your code actually uses.
+
+A hypothetical trimmed package could end up looking like this:
```

Comment:
A hypothetical trimmed package could end up looking like this: ->
Here's what the trimmed package could look like: